src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx

```diff
@@ -641,14 +636,22 @@ function LineplotViz(props: VisualizationProps) {
     [data]
   );
 
-  const defaultDependentAxisRange = useDefaultAxisRange(
+  let defaultDependentAxisRange = useDefaultAxisRange(
```

Comment:
I don't know how strict we are about using constants, but I guess this could be a `const` and then below we could reassign the values within, e.g. `defaultDependentAxisRange.min = ...`  But it would be more verbose

Reply:
Just wanted to say that using `let` for this variable could be avoided by moving this logic into a function, which can use a `let` variable. It has the benefit of keeping the mutable variable out of the scope of the component function.

Reply:
@dmfalke Thank you for your comment! 👍 I also thought of that approach but didn't pursue it as I was not sure of making a function for that specific case only. I will try to do that.

Reply:
@dmfalke I have addressed your suggestion by making a custom hook that wraps the logic. Is it alright for you?

Reply:
@moontrip this looks great. (Sorry for the delayed response!)